### PR TITLE
[CL Hooks]: Add hooks source and clean up source code folder

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
## What is the purpose of the change

This PR adds the source code used for testing in #6859. It also cleans up the testcontracts folder to be more approachable/readable.

This is done in a separate PR to avoid cluttering the base PR since the changes are light & state compatible but the diff is large

### CODE REVIEW

File: x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+//         };
+
+//         let _res = sudo(deps.as_mut(), mock_env(), msg).unwrap();
+     
+//         // Ensure the owner has received 1 token with denom "beforeCreatePosition"
+//         let owner_balance = deps.querier.update_balance("owner", vec![coin(1, "beforeCreatePosition")]);
+//         assert_eq!(owner_balance, Some(vec![coin(1, "beforeCreatePosition")]));
+//     }
+    
+// }
```
Comment (p0mvn) [inline x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs:260]: Do we want this?

**Review by p0mvn — APPROVED** [review]
Sweet

File: x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs (hunk 1 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs:260]: Good point, removed dead code

File: x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        SudoMsg::BeforeCreatePosition {
+            pool_id: _,
+            owner,
+            tokens_provided: _,
+            amount_0_min: _,
+            amount_1_min: _,
+            lower_tick: _,
+            upper_tick: _,
+        } => {
+            let mut response = Response::new();
```
Comment (mattverse) [inline x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/contract.rs:62]: Seems that the following lines are duplicated regardless of which message (hook) it is, can we create a method for these lines?

**Review by mattverse — APPROVED** [review]

File: x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/state.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::Addr;
+use cw_storage_plus::Item;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct State {
+    pub count: i64,
```
[the remark this answers is not in the document]
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/testcontracts/contract-sources/hooks/src/state.rs:9]: This is unused in this case but will likely be important for #6862 so I had just left it there

**Pull request merged**